src/preprocessing/assertion_pipeline.cpp

```diff
@@ -152,82 +206,19 @@ void AssertionPipeline::addSubstitutionNode(Node n, ProofGenerator* pg)
 {
   Assert(d_storeSubstsInAsserts);
   Assert(n.getKind() == Kind::EQUAL);
-  conjoin(d_substsIndex, n, pg);
+  size_t prevNodeSize = d_nodes.size();
+  push_back(n, false, pg);
+  // remember this is a substitution index
+  for (size_t i = prevNodeSize, newSize = d_nodes.size(); i < newSize; i++)
+  {
+    d_substsIndices.insert(i);
+  }
```

Comment:
I'm not seeing why you can do this without having to track it in the proof (as conjoin) would before. Can you clarify?

Reply:
We support proofs in `push_back`, so this is just using that method instead of `conjoin`.

This business of "substitution index" (which doesn't have anything to do with proofs) is slightly more complicated now since equalities corresponding to substitutions can be stored at multiple indices.

Reply:
Ah, I see.